pkg/functionconfig/reader.go

```diff
@@ -36,16 +39,63 @@ func NewReader(parentLogger logger.Logger) (*Reader, error) {
 	}, nil
 }
 
+// Merges codeEntry config with the function config.
+// Config populated values won't get override by codeEntry values.
+// Enriches config with env vars existing only in codeEntry config
 func (r *Reader) Read(reader io.Reader, configType string, config *Config) error {
+	var codeEntryConfigMap, configMap map[string]interface{}
```

Comment:
I know `XConfigMap` was my comment on this but I think I was wong and you were right here. It didn't occur to me that this now might be confusing with k8s' `configMap`.
Your initial suggestion `AsMap` was better here as it prevented this confusion.